libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -690,7 +722,11 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
      * > in your bot's console output, the likely cause is that an async function was used
      * > without using the `await` keyword. Make sure all async functions use await!
      */
-    public async processActivity(req: WebRequest, res: WebResponse, logic: (context: TurnContext) => Promise<any>): Promise<void> {
+    public async processActivity(req: WebRequest|Request, res: WebResponse|ServerUpgradeResponse, logic: (context: TurnContext) => Promise<any>): Promise<void> {
+        if (this.isUpgradeRequest(req) && (req).isUpgradeRequest()) {
+            return this.useWebSocket(req, res as ServerUpgradeResponse, logic);
+        }
```

Comment:
We should just inspect for the `Connection: Upgrade` header. The helper method doesn't make sense, and directly ties us to Restify. 

Checking for the appropriate header will allow us to not ship dependencies on specific web frameworks.

Reply:
Reworked, but still depends on WaterShed until the factory changes are merged.